thingsboard_gateway/tb_utility/tb_utility.py

```diff
@@ -43,8 +43,26 @@ def validate_converted_data(data):
             error = 'deviceName is empty in data: '
         if error is None and not data.get("deviceType"):
             error = 'deviceType is empty in data: '
-        if error is None and data.get("attributes") is None and (data.get("telemetry") is None or (data["telemetry"].get("ts") is not None and len(data["telemetry"].get("values")) == 0)):
-            error = 'No telemetry and attributes in data: '
+
+        if error is None:
+            got_attributes = False
+            got_telemetry = False
+
+            if data.get("attributes") is not None:
+                for entry in data.get("attributes"):
+                    if entry.get("ts") is not None and len(entry.get("values")) > 0:
```

Comment:
Attributes doesn't have an ability to contain timestamp in the message (According to ThingsBoard architecture). This check will be redundant because we always have attributes just like a dictionary. Could you change this?

Reply:
Fixed it. I'm partial to leave the len check on attributes as it is; let me know if you want it removed.